src/utils/transaction.ts

```diff
@@ -160,6 +160,113 @@ export const klimaDaoStaking = async (
+      ...transaction,
+      chainId: plrDaoAsset.chainId,
+      createTimestamp,
+      status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+    }));
+
+    // not native asset and no erc20 approval transaction included
+    if (plrDaoAsset.address && !addressesEqual(plrDaoAsset.address, nativeAssetPerChainId[plrDaoAsset.chainId].address) && transactions.length === 1) {
+      const abi = getContractAbi(ContractNames.ERC20Token);
+      const erc20Contract = sdk.registerContract<ERC20TokenContract>('erc20Contract', abi, plrDaoAsset.address);
+      const approvalTransactionRequest = erc20Contract?.encodeApprove?.(transactions[0].to, amount);
+      if (!approvalTransactionRequest || !approvalTransactionRequest.to) {
+        return { errorMessage: 'Failed build bridge approval transaction!' };
+      }
+
+      const approvalTransaction = {
+        to: approvalTransactionRequest.to,
+        data: approvalTransactionRequest.data,
+        chainId: plrDaoAsset.chainId,
+        value: 0,
+        createTimestamp,
+        status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+      };
+
+      transactions = [approvalTransaction, ...transactions];
+    }
+
+    const abi = getContractAbi(ContractNames.ERC20Token);
+    const erc20Contract = sdk.registerContract<ERC20TokenContract>('erc20Contract', abi, plrDaoAsset.address); // PLR Dao Polygon
+    const plrDaoApprovalTransactionRequest = erc20Contract?.encodeApprove?.(plrDaoMemberNFT.address, offer.receiveAmount); // PLR Dao staking
+    if (!plrDaoApprovalTransactionRequest || !plrDaoApprovalTransactionRequest.to) {
+      return { errorMessage: 'Failed build bridge approval transaction!' };
+    }
+
+    const plrDaoApprovalTransaction = {
+      to: plrDaoApprovalTransactionRequest.to,
+      data: plrDaoApprovalTransactionRequest.data,
+      chainId: plrDaoAsset.chainId,
+      value: 0,
+      createTimestamp,
+      status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+    };
+
+    const plrDaoStakingAbi = [
+      "function stake(uint256 value)",
+    ];
+    const plrDaoStakingContract = sdk.registerContract<{ encodeStake: (amount: BigNumberish) => TransactionRequest }>('plrDaoStakingContract', plrDaoStakingAbi, plrDaoMemberNFT.address); // PLR Dao Polygon
+    const plrDaoStakeTransactionRequest = plrDaoStakingContract.encodeStake?.(offer.receiveAmount); // PLR Dao staking
+    if (!plrDaoStakeTransactionRequest || !plrDaoStakeTransactionRequest.to) {
+      return { errorMessage: 'Failed build bridge approval transaction!' };
+    }
+
+    const plrDaoStakinglTransaction = {
+      to: plrDaoStakeTransactionRequest.to,
+      data: plrDaoStakeTransactionRequest.data,
+      chainId: plrDaoAsset.chainId,
+      value: 0,
+      createTimestamp,
+      status: CROSS_CHAIN_ACTION_STATUS.UNSENT,
+    };
+
+    transactions = [...transactions, plrDaoApprovalTransaction, plrDaoStakinglTransaction];
+
+    if (receiverAddress && receiverAddress != sdk.state.accountAddress) {
```

Comment:
Does TS handle `!=` as opposed to `!==`?

Reply:
FYI: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_inequality